packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1109,6 +1109,14 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   /// It is an error to call [setState] unless [mounted] is true.
   bool get mounted => _element != null;
 
+  /// Whether this object is active in the tree.
+  ///
+  /// Usually starts with true when calling [initState].
+  ///
+  /// If [widget] or ancestor widget has [GlobalKey], then active is changed in
+  /// [deactivate] and [activate].
+  bool get active => _element?._lifecycleState == _ElementLifecycle.active;
```

Comment:
what's the use case for this getter?

Reply:
Unless there's a good use case I'd rather we didn't add this. In general exposing lifecycles like this is not necessary and may encourage poor practices.

Reply:
Maybe it has been too long and we have forgotten some details.

This is so that updateKeepAlive can be executed at the right time. If Material is calling dispose, but InkWell is inactive, then InkHighlight.remove is executed. If we don't know if it's active at this time, this won't work.

Before this I tried manually recording active inside inkwell.state, but I felt it was very ugly...

btw, sorry I ignored the reply.